### PR TITLE
Drop support php 7, add php 8.2 support

### PR DESCRIPTION
**Pull request by fezfez**

### CODE REVIEW

**Comment by fezfez** [comment]
@Ocramius i dont know how to deal with dep-on-root... have you an idea ?
```
/var/www # composer update
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires laminas/laminas-mail ~2.19.0 -> satisfiable by laminas/laminas-mail[2.19.0].
    - laminas/laminas-mail 2.19.0 requires laminas/laminas-mime ^2.10.0 -> satisfiable by laminas/laminas-mime[2.10.0] from composer repo (https://repo.packagist.org) but laminas/laminas-mime is the root package and cannot be modified. See https://getcomposer.org/dep-on-root for details and assistance.

```

File: composer.json [hunk 1]
```diff
@@ -22,13 +22,13 @@
...
-        "php": "^7.3 || ~8.0.0 || ~8.1.0",
+        "php": "~8.0.0 || ~8.1.0 || ~8.2.0",
         "laminas/laminas-stdlib": "^2.7 || ^3.0"
     },
     "require-dev": {
-        "laminas/laminas-coding-standard": "~2.2.1",
-        "laminas/laminas-mail": "^2.12",
-        "phpunit/phpunit": "^9.5"
+        "laminas/laminas-coding-standard": "~2.4.0",
+        "laminas/laminas-mail": "~2.19.0",
```
Comment (Ocramius) [inline composer.json]: Hmm, this bi-directional dependency is problematic - not sure how to solve this, other than ignoring platform requirements during installation

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (Ocramius) [reply composer.json]: @gsteel you dealt with a bunch of these recently: have any suggestions?

Comment (gsteel) [reply composer.json]: We'll have to checkout `mime` as `2.99.x` in `pre-install.sh` like this:
https://github.com/laminas/laminas-view/blob/2.25.x/.laminas-ci/pre-install.sh

Comment (fezfez) [reply composer.json]: i got the same error, did i miss something ?

```
- git co -b 2.99.x
Switched to a new branch '2.99.x'
- composer update
Loading composer repositories with package information
Info from https://repo.packagist.org: #StandWithUkraine
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

Problem 1
- Root composer.json requires laminas/laminas-mail ~2.19.0 -> satisfiable by laminas/laminas-mail[2.19.0].
    - laminas/laminas-mail 2.19.0 requires laminas/laminas-mime ^2.10.0 -> satisfiable by laminas/laminas-mime[2.10.0] from composer repo (https://repo.packagist.org) but laminas/laminas-mime is the root package and cannot be modified. See https://getcomposer.org/dep-on-root for details and assistance.

```

Comment (Ocramius) [reply composer.json]: Can you try pushing that? It may be composer trying to require a stable version in your environment...

Comment (fezfez) [reply composer.json]: I am in docker env and git wasn't installed.

@Ocramius seem to be good, thanks @gsteel !

File: composer.lock [hunk 2]
```diff
@@ -401,67 +401,6 @@
             ],
             "time": "2022-10-14T13:05:29+00:00"
         },
-        {
-            "name": "laminas/laminas-mime",
```
Comment (Ocramius) [inline composer.lock]: Ha, funny how this was in the lock file!

File: composer.json (hunk 1 above)
Comment (Ocramius) [reply composer.json]: @fezfez this should be `^2.19.0` BTW: `~2.19.0` is too restrictive

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (fezfez) [reply composer.json]: done

**Review by Ocramius — APPROVED** [review]
Thanks @fezfez!

**Pull request merged**